### PR TITLE
Add Twitter client

### PR DESCRIPTION
**Pull request by vigneshc**
- Add TwitterClientCore, .net core client to listen to twitter and send data to eventhub.

### CODE REVIEW

File: DataGenerators/TwitterClientCore/TwitterStream.cs [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+            // unique request details
+            var oauth_nonce = Convert.ToBase64String(new ASCIIEncoding().GetBytes(DateTime.Now.Ticks.ToString()));
+            var oauth_timestamp = Convert.ToInt64(
+                (DateTime.UtcNow - new DateTime(1970, 1, 1, 0, 0, 0, 0, DateTimeKind.Utc))
+                    .TotalSeconds).ToString();
+
+            var resource_url = "https://stream.twitter.com/1.1/statuses/filter.json";
+
+            // create oauth signature
+            var baseString = string.Format(
```
Comment (viskri-msft) [inline DataGenerators/TwitterClientCore/TwitterStream.cs]: nit: use string interpolation so we can be consistent everywhere.

File: DataGenerators/TwitterClientCore/TwitterStream.cs [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+
+            string oauth_signature;
+            using (var hasher = new HMACSHA1(ASCIIEncoding.ASCII.GetBytes(compositeKey)))
+            {
+                oauth_signature = Convert.ToBase64String(
+                hasher.ComputeHash(ASCIIEncoding.ASCII.GetBytes(baseString)));
+            }
+
+            // create the request header
+            var authHeader = string.Format(
```
Comment (viskri-msft) [inline DataGenerators/TwitterClientCore/TwitterStream.cs]: ditto

File: DataGenerators/TwitterClientCore/TwitterStream.cs [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+            HttpWebRequest request = (HttpWebRequest)WebRequest.Create(resource_url);
+            request.Headers.Add("Authorization", authHeader);
+            request.Method = "POST";
+            request.ContentType = "application/x-www-form-urlencoded";
+            request.PreAuthenticate = true;
+            request.AllowWriteStreamBuffering = true;
+            request.CachePolicy = new System.Net.Cache.RequestCachePolicy(System.Net.Cache.RequestCacheLevel.BypassCache);
+
+            // bail out and retry after 5 seconds
+            var tresponse = request.GetResponseAsync();
```
Comment (viskri-msft) [inline DataGenerators/TwitterClientCore/TwitterStream.cs]: nit: camelCase

File: DataGenerators/TwitterClientCore/App.config [hunk 4]
```diff
@@ -0,0 +1,18 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<configuration>
+  
+  <appSettings>
+    <!-- Twitter Streaming API settings.  Replace Twitter OAuth settings with your own -->
+    <add key="oauth_consumer_key" value="**REPLACE WITH KEY**" />
+    <add key="oauth_consumer_secret" value="**REPLACE WITH KEY**" />
+    <add key="oauth_token" value="**REPLACE WITH KEY**" />
+    <add key="oauth_token_secret" value="**REPLACE WITH KEY**" />
```
[the remark this answers is not in the document]
Comment (vigneshc) [reply DataGenerators/TwitterClientCore/App.config]: Readme.md file has link to twitter developer documentation page that has details about getting these keys.

**Pull request closed without merging**